### ISSUE TITLE
Default to vcFormat if an `.editorconfig` containing vcFormat entries was found nearer than any `.clang-format` file

### ISSUE THREAD

**Issue by Colengms** [microsoft/vscode-cpptools#7929]
Currently, the `Default` setting of `Cpp.formatting` is the same as `clangFormat`.

This would redefine `Default` to be dynamic selection of either clang-format or vcFormat depending on whether a `.clang-format` file or an `.editorconfig` file (containing vcFormat entries) are found, or which is nearer the file to be formatted.

Defaulting to clang-format if both are found, and only using an `.editorconfig` if it contains vcFormat entries, would avoid changing the behavior for users who either intended to use clangFormat, are configured for both (so it shouldn't matter which is used), or might otherwise not appreciate having their formatting arbitrarily changed.  The existance of the vcFormat entries and no `.clang-format` file (at better or equivalent distance), would seem a reasonable indication that vcFormat is intended to be used.

Instead of sending the value of `Cpp.formatting` to the native side with other settings, it could be sent only per formatting request, allowing it to be varied on the TypeScript side when set to `Default`.

**Comment by sean-mcmanus** [comment]
I don't think the "distance" should be a factor.

**Comment by Colengms** [comment]
@sean-mcmanus .  Considering these two scenarios:

```
workspace_root\.clang-format
workspace_root\subdirectory1\.editorconfig
workspace_root\subdirectory1\subdirectory2\source.cpp
```

```
workspace_root\.editorconfig
workspace_root\subdirectory1\.clang-format
workspace_root\subdirectory1\subdirectory2\source.cpp
```

By "distance", I'm suggesting the first example would use `.editorconfig` and the second example use `.clang-format`, based on being nearer the source file.  

Are you suggesting that both examples should use `.clang-format` or both should use `.editorconfig`?

**Comment by sean-mcmanus** [comment]
Okay, changed my mind, distance seems okay to consider.

**Comment by Colengms** [comment]
This is addressed in 1.6.0-insiders.
